### DISCUSSION TITLE
Is this script a good fit?

### DISCUSSION THREAD

**Discussion by n8jadams** [angristan/wireguard-install#466] [Help] [answered]
Hey y'all. I already have a VPN set up in my home network with my unraid server, which has a nice GUI to manage all of this.

But I have an old macbook with ubuntu on it, and I'm looking to use it as a dedicated proxy to my parent's network. I also have a duckdns domain already set up and a cron-job updating in the case of dynamic ip.

As far as I understand, at this point I should be able to:
* In my parents router settings, assign a constant LAN IP to my macbook
* Again, in my parents router settings, set up port forwarding, forwarding udp 51820 to the LAN IP of the macbook on port 51820
* Run this script, setting `SERVER_PUB_IP` to be my duckDNS subdomain.

Then I should be able to just create some client peer conf files, load them in a wireguard client GUI from anywhere with an internet connection, and then ssh into my ubuntu-macbook, open any LAN services on that network, or run applications that will see my ip as my parent's ip address. (Remote tunnel access I believe is the term.)

Am I understanding this correctly? Is this script a good fit for my use case? All of the language in the readme is around VPSs. I've tried a bunch of different things and I figured I'd ask now before I put any more time into this project.

**Comment by pothi** [comment, marked as answer]
You can install this script on a macbook running ubuntu and let it act as a proxy for other computers connected to it.

**Reply by n8jadams** [reply]
Are the steps that I listed comprehensive and accurate?

**Comment by n8jadams** [comment]
Success! Set the IPv4 address as my duckdns domain (no https:// no port), then server wireguard port to 51820. Set up udp port forwarding to my device. Worked like a charm. Great script. Thanks for this!
